BungeeCord-Patches/0056-Save-memory-on-scoreboard-teams.patch

```diff
@@ -0,0 +1,107 @@
+From 79a63969805be4e5d444eb4f7360141de14176d7 Mon Sep 17 00:00:00 2001
+From: Mark Vainomaa <mikroskeem@mikroskeem.eu>
+Date: Fri, 8 May 2020 21:39:39 +0300
+Subject: [PATCH] Save memory on scoreboard teams
+
+
+diff --git a/api/src/main/java/net/md_5/bungee/api/score/Team.java b/api/src/main/java/net/md_5/bungee/api/score/Team.java
+index 39f81fd1..093784f1 100644
+--- a/api/src/main/java/net/md_5/bungee/api/score/Team.java
++++ b/api/src/main/java/net/md_5/bungee/api/score/Team.java
+@@ -17,8 +17,8 @@ public class Team
+     private String prefix;
+     private String suffix;
+     private byte friendlyFire;
+-    private String nameTagVisibility;
+-    private String collisionRule;
++    private NametagVisibility nameTagVisibility; // Waterfall - save memory on scoreboard teams
++    private CollisionRule collisionRule; // Waterfall - save memory on scoreboard teams
+     private int color;
+     private Set<String> players = new HashSet<>();
+ 
+@@ -32,6 +32,82 @@ public class Team
+         players.add(name.intern());
+     }
+ 
++    // Waterfall start - save memory on scoreboard teams
++    @Deprecated
++    public String getNameTagVisibility() {
++        return nameTagVisibility.getValue();
++    }
++
++    @Deprecated
++    public void setNameTagVisibility(String value) {
++        NametagVisibility enumValue;
++        if (value == null) {
```

Comment:
I don't see a reason to add null-safety here. Its not API and undocumented behaviour, just set enumValue to null, same for collisonrule and if value was not found.

Reply:
Right, will do so.

Reply:
I edited my comment.